src/libServer/IsolatedServer.cpp

```diff
@@ -769,16 +769,26 @@ Json::Value IsolatedServer::CreateTransactionEth(EthFields const& fields,
     ret["TranID"] = txHash.hex();
     ret["Info"] = "Txn processed";
     WebsocketServer::GetInstance().ParseTxn(twr);
-    LOG_GENERAL(INFO, "Processing On the isolated server completed");
-    return ret;
-
+    LOG_GENERAL(
+        INFO,
+        "Processing On the isolated server completed. Minting a block...");
   } catch (const JsonRpcException& je) {
     LOG_GENERAL(INFO, "[Error]" << je.what() << " Input JSON: NA");
     throw je;
   } catch (exception& e) {
     LOG_GENERAL(INFO, "[Error]" << e.what() << " Input code: NA");
     throw JsonRpcException(RPC_MISC_ERROR, "Unable to Process");
   }
+
+  // Double create a block to make sure TXs are 'flushed'
+  // This will make sure the block height advances, the
+  // TX can be found in a block etc.
+  if (m_timeDelta == 0) {
```

Comment:
Sorry, I don't fully understand why we call it twice. Does it create genesis block on the first run? Is my understanding correct?

Reply:
I believe there is an issue around height 1 which causes this to be neccessary to satisfy metamask. The block height starts at 1 and if you submit a tx and mint a block it goes into block 1 which is not how the chain works usually as that is like injecting into something that is finalised.

For convenience I just mint 2 blocks but if if bothers you I can write a workaround so it only needs to mint 1.

Reply:
It's still IsolatedServer so I think these type of workarounds are ok. Thanks for clarification.